### PR TITLE
Added support show/demo mode switch in config

### PR DESCRIPTION
**Pull request by Princess-of-Sleeping**

### CODE REVIEW

File: vita3k/modules/SceVshBridge/SceVshBridge.cpp [hunk 1]
```diff
@@ -806,15 +806,15 @@ EXPORT(int, vshSysconIsDownLoaderMode) {
 }
 
 EXPORT(int, vshSysconIsIduMode) {
-    return UNIMPLEMENTED();
+    return emuenv.cfg.current_config.demo_mode == true ? 1 : 0;
```
Comment (Zangetsu38) [inline vita3k/modules/SceVshBridge/SceVshBridge.cpp:809]: you can remove == true,  no needed ^^

File: vita3k/modules/SceVshBridge/SceVshBridge.cpp [hunk 2]
```diff
@@ -806,15 +806,15 @@ EXPORT(int, vshSysconIsDownLoaderMode) {
...
 }
 
 EXPORT(int, vshSysconIsMCEmuCapable) {
     return UNIMPLEMENTED();
 }
 
 EXPORT(int, vshSysconIsShowMode) {
-    return UNIMPLEMENTED();
+    return emuenv.cfg.current_config.show_mode == true ? 1 : 0;
```
Comment (Zangetsu38) [inline vita3k/modules/SceVshBridge/SceVshBridge.cpp:817]: same ^^

**Comment by Zangetsu38** [comment]
set code inside setting dialog for it, just copy past exist code exist in system tab for pstv mode and change boot + name etc biy your new bool 👌

File: vita3k/modules/SceVshBridge/SceVshBridge.cpp (hunk 1 above)
Comment (Zangetsu38) [reply vita3k/modules/SceVshBridge/SceVshBridge.cpp:809]:
```suggestion
    return emuenv.cfg.current_config.demo_mode  ? 1 : 0;
```
sorry, i have wanted that

Comment (Princess-of-Sleeping) [reply vita3k/modules/SceVshBridge/SceVshBridge.cpp]: done

**Comment by Zangetsu38** [comment]
change commit name too is invalide here
change to:
config: Added support show/demo mode switch

File: vita3k/config/include/config/state.h [hunk 3]
```diff
@@ -173,6 +173,8 @@ struct Config : YamlLoader {
         int modules_mode = ModulesMode::AUTOMATIC;
         std::vector<std::string> lle_modules = {};
         bool pstv_mode = false;
+        bool show_mode = false;
+        bool demo_mode = false;
         bool ngs_enable = true;
```
[the remark this answers is not in the document]
Comment (Princess-of-Sleeping) [reply vita3k/config/include/config/state.h:178]: If you don't add this line you will get a build error.

[the remark this answers is not in the document]
Comment (Zangetsu38) [reply vita3k/config/include/config/state.h:178]: ??? is just custom config, how that can caused error, all value is not here

[the remark this answers is not in the document]
Comment (Princess-of-Sleeping) [reply vita3k/config/include/config/state.h:178]:
```
重大度レベル	コード	説明	プロジェクト	ファイル	行	抑制状態
エラー	C2039	'demo_mode': 'Config::CurrentConfig' のメンバーではありません	modules	D:\desktop\Vita3K-master\Vita3K\vita3k\modules\SceVshBridge\SceVshBridge.cpp	809	
エラー	C2039	'show_mode': 'Config::CurrentConfig' のメンバーではありません	modules	D:\desktop\Vita3K-master\Vita3K\vita3k\modules\SceVshBridge\SceVshBridge.cpp	817	
```

File: vita3k/modules/SceVshBridge/SceVshBridge.cpp [hunk 4]
```diff
@@ -806,15 +806,15 @@ EXPORT(int, vshSysconIsDownLoaderMode) {
...
+    return emuenv.cfg.current_config.demo_mode ? 1 : 0;
 }
 
 EXPORT(int, vshSysconIsMCEmuCapable) {
     return UNIMPLEMENTED();
 }
 
 EXPORT(int, vshSysconIsShowMode) {
-    return UNIMPLEMENTED();
+    return emuenv.cfg.current_config.show_mode ? 1 : 0;
```
Comment (Zangetsu38) [inline vita3k/modules/SceVshBridge/SceVshBridge.cpp:817]: same, remove current_config

File: vita3k/modules/SceVshBridge/SceVshBridge.cpp [hunk 5]
```diff
@@ -806,15 +806,15 @@ EXPORT(int, vshSysconIsDownLoaderMode) {
 }
 
 EXPORT(int, vshSysconIsIduMode) {
-    return UNIMPLEMENTED();
+    return emuenv.cfg.current_config.demo_mode ? 1 : 0;
```
[the remark this answers is not in the document]
Comment (Princess-of-Sleeping) [reply vita3k/modules/SceVshBridge/SceVshBridge.cpp]: done.

**Review by Zangetsu38 — APPROVED** [review]

**Pull request merged**